transformations/negate_strengthen/transformation.py

```diff
@@ -0,0 +1,462 @@
+        table['head_id'] = [letters2word[i] for i in table['head_id']]
+        if self.verbose:
+            print(table)
+
+        return table
+
+
+    def negation_rules(self, tgx, itemdict, text, pos, sentid2tid, method=[], \
+        num_tries=2, curr_try=0):
+
+        curr_try += 1
+        edit_id = tgx
+
+        if self.verbose:
+            print('before {}, actual {}, after {}'.format(pos[tgx-2], pos[tgx-1], pos[tgx]))
+        if curr_try > num_tries:
+            if self.verbose:
+                print('Max {} tries hit...'.format(num_tries))
+            method.append(None)
+            edit_id = None
+            text = None
+        elif pos[tgx-1][0:2] =='VB':
+            # if actual word is a modal type
+            if self.wnl.lemmatize(text[tgx-1],'v') in MODAL_TYPE_LEMMA \
+            or (pos[tgx-1] in ['MD']):
+                # if word is last word
+                if tgx>=len(text):
+                    method.append('VB_1_1')
+                    text[tgx-1] =  "<edit>not</edit> " + text[tgx-1]
+                # if word after is determiner of sorts
+                elif pos[tgx][0:2] in ['CD', 'DT']:
+                    method.append('VB_1_2')
+                    text[tgx] = "<edit>no</edit>"
+                # if word after is noun or description of a noun
+                elif pos[tgx][0:2] in ['JJ', 'NN', 'PR', 'WP', 'WD', 'DT']:
+                    method.append('VB_1_3')
+                    text[tgx-1] = text[tgx-1] + " <edit>not</edit>"
+                # if word after is VB
+                elif pos[tgx][0:2] =='VB':
+                    method.append('VB_1_4')
+                    text[tgx-1] = text[tgx-1] + " <edit>no</edit>"
+            # if word is the first word
+            elif tgx-1==0:
+                method.append('VB_2_1')
+                text[tgx-1] = "<edit>Not</edit> " + text[tgx-1].lower()
+            # if word before target is noun type
+            elif pos[tgx-2][0:2] in ['NN', 'PR', 'WP', 'WD', 'DT']:
+                method.append('VB_3_1')
+                text[tgx-1] = "<edit>did not</edit> " + self.wnl.lemmatize(text[tgx-1],'v')
+            # if word is last word
+            elif tgx>=len(text):
+                method.append('VB_4_1')
+                text[tgx-1] =  "<edit>not</edit> " + text[tgx-1]
+            # if word before target is AUX | if word after target is IN/TO
+            elif self.wnl.lemmatize(text[tgx-2],'v') in MODAL_TYPE_LEMMA \
+            or (pos[tgx-2] in ['MD'])\
+            or (pos[tgx] in ['IN', 'TO']):
+                method.append('VB_5_1')
+                text[tgx-1] = "<edit>not</edit> " + text[tgx-1]
+            else:
+                if self.verbose:
+                    print('Unable to find VB_X_X rule...')
+                method.append(None)
+                edit_id = None
+                text = None
+                # text[tgx-1] = "<edit>did not</edit> " + self.wnl.lemmatize(text[tgx-1],'v')
```

Comment:
I can see a few commented codes, could you please help to remove those?

Reply:
Ok, done!